modules/tensor_mechanics/src/materials/SingleVariableReturnMappingSolution.C

```diff
@@ -44,13 +42,24 @@ validParams<SingleVariableReturnMappingSolution>()
                         "algorithm. Also use same old defaults for relative_tolerance, "
                         "absolute_tolerance, and max_its.");
 
+  // diagnostic output parameters
+  MooseEnum internal_solve_output_enum("never on_error always", "on_error");
+  params.addParam<MooseEnum>("internal_solve_output",
+                             internal_solve_output_enum,
+                             "When to output Newton solve information");
+  params.addParam<bool>(
+      "output_iteration_info", false, "Set true to output full Newton iteration history");
```

Comment:
I'm OK with these names, but I'm not behind them 100%.  Since we're changing things anyway, and it's not really going to affect anyone by removing `output_iteration_info` (since I'm pretty sure nobody is using it because they would be getting an insane amount of information), how about this:
```
internal_solve_output_on: "When to output internal Newton solve information"
internal_solve_full_iteration_history:  "Set true to output full internal Newton iteration history at times determined by `internal_solve_output_on`. If false, only a summary is output."
```